app/models/katello/glue/pulp/rpm.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  module Glue::Pulp::Rpm
+    extend ActiveSupport::Concern
+    include LazyAccessor
+
+    included do
+      lazy_accessor :pulp_facts, :initializer => :backend_data
+
+      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
+                    :changelog, :group, :size, :url, :build_time, :group,
+                    :initializer => :pulp_facts
+    end
+
+    def requires
+      if pulp_facts['requires']
+        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
+      else
+        []
+      end
```

Comment:
So there are few things here.

1.  Multiple returns within a function being discouraged

Are you confused by this?  Do you disagree with this?  I swear you have
commented on my code telling me to only return once.

2.  Does my version qualify as having multiple returns

I don't think it does qualify.  The difference is subtle.  The reason I say
this is that the if statement results in a single value.  You could rewrite
it to be more obvious why it doesn't quality:

```
   to_ret = if true
                    1
                else
                     2
                end
   to_ret
```

3.  Does your version qualify as having multiple returns.

 Yes, there is an explicit version and a non-explicit version


On Mon, Aug 3, 2015 at 6:09 PM, Eric D Helms <notifications@github.com>
wrote:

> In app/models/katello/glue/pulp/rpm.rb
> <https://github.com/Katello/katello/pull/5326#discussion_r36137657>:
>
> > +    include LazyAccessor
> > +
> > +    included do
> > +      lazy_accessor :pulp_facts, :initializer => :backend_data
> > +
> > +      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
> > +                    :changelog, :group, :size, :url, :build_time, :group,
> > +                    :initializer => :pulp_facts
> > +    end
> > +
> > +    def requires
> > +      if pulp_facts['requires']
> > +        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
> > +      else
> > +        []
> > +      end
>
> I am still not seeing the 'why' I guess? And in this case there is one
> explicit return statement in the guard clause?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326/files#r36137657>.
>